enhancements/sig-architecture/224-event-based-manifestwork/README.md

```diff
@@ -0,0 +1,1042 @@
+
+```JSON5
+// Context Attributes,
+{
+  "specversion": "1.0",
+  "type": "io.open-cluster-management.works.v1alpha1.manifestbundle.status.update_request",
+  "source": "cluster1-work-agent",
+  "id": "2623db30-32ee-4309-b464-b46f42737af3",
+  "time": "2023-07-19T03:03:11.548189454Z",
+  "datacontenttype": "application/json",
+}
+// Extensions
+{
+  "resourceid": "a52adbe8-b6f2-52c8-9378-c4f544502fb7",
+  "resourceversion": 1
+}
+// Data
+{
+"conditions": [
+  {
+    "type": "Applied",
+    "status": "True",
+    "observedGeneration": 1,
+    "lastTransitionTime": "2023-07-19T03:01:11Z",
+    "reason": "AppliedManifestWorkComplete",
+    "message": "Apply manifest work complete"
+  },
+  {
+    "type": "Available",
+    "status": "False",
+    "observedGeneration": 3,
+    "lastTransitionTime": "2023-07-19T04:45:46Z",
+    "reason": "ResourcesNotAvailable",
+    "message": "1 of 1 resources are not available"
+  }
+],
+"resourceStatus": [{
+    "resourceMeta": {
+        "ordinal": 0,
+        "group": "apps",
+        "version": "v1",
+        "kind": "Deployment",
+        "resource": "deployments",
+        "name": "busybox-48150",
+        "namespace": "default"
+    },
+    "statusFeedback": {
+        "values": [
+          {
+            "name": "status",
+            "fieldValue": {
+            "type": "JsonRaw",
+            "jsonRaw": "{\"availableReplicas\":1,\"conditions\":[{\"lastTransitionTime\":\"2023-07-19T03:01:11Z\",\"lastUpdateTime\":\"2023-07-19T03:01:14Z\",\"message\":\"ReplicaSet \\\"busybox-48150-9cbccc9c7\\\" has successfully progressed.\",\"reason\":\"NewReplicaSetAvailable\",\"status\":\"True\",\"type\":\"Progressing\"},{\"lastTransitionTime\":\"2023-07-19T04:01:15Z\",\"lastUpdateTime\":\"2023-07-19T04:01:15Z\",\"message\":\"Deployment has minimum availability.\",\"reason\":\"MinimumReplicasAvailable\",\"status\":\"True\",\"type\":\"Available\"}],\"observedGeneration\":1,\"readyReplicas\":1,\"replicas\":1,\"updatedReplicas\":1}"
+            }
+          }
+        ]
+    },
+    "conditions": [
+      {
+        "type": "Applied",
+        "status": "True",
+        "lastTransitionTime": "2023-07-19T03:01:11Z",
+        "reason": "AppliedManifestComplete",
+        "message": "Apply manifest complete"
+      },
+      {
+        "type": "Available",
+        "status": "False",
+        "lastTransitionTime": "2023-07-19T04:45:46Z",
+        "reason": "ResourceNotAvailable",
+        "message": "Resource is not available"
+      },
+      {
+        "type": "StatusFeedbackSynced",
+        "status": "True",
+        "lastTransitionTime": "2023-07-19T03:01:11Z",
+        "reason": "StatusFeedbackSynced",
+        "message": ""
+      }
+    ]}]
+}
+```
+
+### Spec Resync Event
+
+#### Event Scenarios
+
+1. After a work agent restarts, it sends a spec resync event with empty payload to the broker to synchronize all resources to avoid missing events during the restart period.
```

Comment:
Does the agent reapply manifests periodically or does it just apply it once after receiving the command from the broker?

Reply:
this depends on the version of the resource, the agent will compare the resource version, If the version of the resource received by the agent is greater than the version of the resource maintained by the agent, the agent will reapply the resource to update it

Reply:
In general, if there is no change by the user, it will not constantly reapply it even if the cluster state has changed?

Reply:
Agent has a MQTT interface to get the CR. Thereafter, it should be business as usual. And then it has MQTT interface to relay back the status (or CR update)

Reply:
for work agent, it maintains the spec of the manifests on the managed and reports the status of the manifests, if only the status is changed, the work agent will not reapply the manifests, but it will report the manifests new status in the manifestwork